docs/DEVELOPMENT.md

```diff
@@ -112,124 +129,46 @@ To run the tests, we need a test db.
-access to push the image to the registry.
-
-Make sure you are logged into the cluster and you are in the
-`hub/api` directory before running the following command
-
-```bash
-ko apply -f config/
-```
-
-The command above will create an image and push it to registry pointed by  `KO_DOCKER_REPO` and then deploy the `api` and `db` container on the cluster.
-
-**NOTE:** Ensure that the image is **publicly** available otherwise deployment will fail. The pod status will give `Error: ImagePullBackOff`
-
-Watch the pods until both of them are running.
-```
-kubectl get pods -o wide -w
-```
-
-To create tables and to populate, we need to run the migration. Run the following command:
-
-```
-ko apply -f config/db-migration/14-db-migration.yaml
-```
-
-The command above will create a migration image, push to registry pointed by `KO_DOCKER_REPO` and run a `job` that performs the migration.
-
-Check the logs using
-
-```bash
-kubectl logs job/db-migration
-```
-
-Wait until the migration log shows
-
-> DB initialisation successful !!
-
-### Expose API service
-
-To expose the api service to be accessible outside cluster, run the following command based the `k8s` cluster you are using:
-
-#### OpenShift
-
-```bash
-oc apply -f config/openshift/
-```
-This will expose the api service as a `NodePort` service and create a `route` that can be used to access the api.
-
-you can get the route to api service using
-
-```bash
-oc get routes api --template='https://{{ .spec.host }} '
+  go test -p 1 -count=1 -v ./pkg/...
 ```
 
-#### GKE
-
+To re-generate the golden files use the below command
 ```bash
-kubectl apply -f  config/gke/21-api-service.yaml
+go test ./pkg/... -test.update-golden
 ```
 
-This will expose the deployment as `LoadBalancer` service. You can access the api service using `LoadBalancer Ingress` and `Port`.
+**NOTE:** `tests` use the database configurations from [test/config/env.test][env-test-file]
 
-```bash
-kubectl describe svc api-load
-```
-Use the `LoadBalancerIngress:Port` eg. `http://35.223.169.99:8000`
+### Creating JWT for testing
 
+To create a JWT, create a HTML file adding below code
+```html
+<!DOCTYPE html>
+<html>
 
-#### Test API Service
+<body>
+  <a href="https://github.com/login/oauth/authorize?client_id=<Add Client ID here>">
+    Login with github
+  </a>
+</body>
 
-Run `curl http://<ip>:port/categories` api to make sure
-everything is working fine.
+</html>
+````
+Add your OAuth Client ID from [.env.dev][env-dev] in HTML file in place of `<Add Client ID here>` and Save it.
 
+Open the HTML file, click on `Login with github` It will redirect you to GitHub. Once you authorize, it will redirect you to url localhost.
 
-#### Redeploy API Service
+for ex. `http://localhost:8080/?code=32d4a0b4eb6e9fbea731`
 
-**To redeploy api service after changes run the following command:**
+Use the `code` from url in `/auth/login` API. It will create a user in db and return a JWT. 
```

Comment:
Can we change this to 
By making a `POST` call to the `/auth/login` API, use the `code` from url in by passing it as params?

Reply:
that would be okay I guess we are just letting the user know what he has to access. the user will have to look at the API definition :D before it is accessed